NiBAx/plugins/SmileGAN/SmileGAN.py

```diff
@@ -0,0 +1,292 @@
+            self.thread.start()
+
+
+    def PopulateHue(self):
+        #add the list items to comboBoxHue
+        datakeys = self.datamodel.GetColumnHeaderNames()
+        datatypes = self.datamodel.GetColumnDataTypes()
+        categoryList = ['Sex','Study','A','T','N','PIB_Status'] + [k for k,d in zip(datakeys, datatypes) if d.name=='category']
+        categoryList = list(set(categoryList).intersection(set(datakeys)))
+        self.ui.comboBoxHue.clear()
+        self.ui.comboBoxHue.addItems(categoryList)
+
+
+    def plotSPAREs(self, useExistingSPAREs=True):
+        # Plot data
+        if self.ui.stackedWidget.currentIndex() == 0:
+            return
+        self.plotCanvas.axes.clear()
+        plotOptions = {'HUE': self.ui.comboBoxHue.currentText()}
+
+        if useExistingSPAREs:
+            print(self.SPAREs[plotOptions['HUE']].value_counts())
+            kws = {"s": 20}
+            sns.scatterplot(x='SPARE_AD', y='SPARE_BA', data=self.SPAREs,
+                        ax=self.plotCanvas.axes, linewidth=0, hue=plotOptions['HUE'],
+                        facecolor=(0.5, 0.5, 0.5, 0.5), **kws) 
+        else:
+            kws = {"s": 20}
+            sns.scatterplot(x='SPARE_AD', y='SPARE_BA', data=self.SPAREs,
+                        ax=self.plotCanvas.axes, linewidth=0,
+                        facecolor=(0.5, 0.5, 0.5, 0.5), legend=None)
+            
+
+
+        sns.despine(ax=self.plotCanvas.axes, trim=True)
+        self.plotCanvas.axes.set(ylabel='SPARE-BA', xlabel='SPARE-AD')
+        self.plotCanvas.axes.get_figure().set_tight_layout(True)
+        self.plotCanvas.draw()
+
+
+    def OnAddToDataFrame(self):
+        print('Adding SPARE-* scores to data frame...')
+        self.datamodel.data.loc[:,'SPARE_AD'] = self.SPAREs['SPARE_AD']
+        self.datamodel.data.loc[:,'SPARE_BA'] = self.SPAREs['SPARE_BA']
+        self.datamodel.data_changed.emit()
+        self.OnShowSPAREs()
+
+
+    def OnShowSPAREs(self):
+        allHues = [self.ui.comboBoxHue.itemText(i) for i in range(self.ui.comboBoxHue.count())]
+
+        self.SPAREs = self.datamodel.data[['SPARE_BA', 'SPARE_AD'] + allHues]
+        self.ui.stackedWidget.setCurrentIndex(1)
+        self.ui.comboBoxHue.setVisible(True)
+        self.plotSPAREs()
+
+
+    def OnDataChanged(self):
+        # Set `Show SPARE-* from data` button to visible when SPARE-* columns
+        # are present in data frame
+        self.ui.stackedWidget.setCurrentIndex(0)
+        if ('SPARE_BA' in self.datamodel.GetColumnHeaderNames() and
+            'SPARE_AD' in self.datamodel.GetColumnHeaderNames()):
+            self.ui.show_SPARE_scores_from_data_Btn.setEnabled(True)
+            self.ui.show_SPARE_scores_from_data_Btn.setStyleSheet("background-color: rgb(230,230,255)")
+        else:
+            self.ui.show_SPARE_scores_from_data_Btn.setEnabled(False)
+        
+        self.PopulateHue()
+
+
+class BrainAgeWorker(QtCore.QObject):
```

Comment:
This class also looks like a duplicate of the one in computeSpares.py. Can we not re-use that?

Reply:
Sorry for the confusion. I am adding a Smile-GAN plugin with Ahmed step by step now, so it is still in progress and it is now mostly copied from compute-spare.

Reply:
@zhijian-yang Is this PR still a work in progress? If yes, we can convert this into a draft.
Also, if you want to re-use code from ComputeSpares class, a better approach is to derive from that class.